src/Cron/HoursField.php

```diff
@@ -167,46 +166,42 @@ public function increment(DateTimeInterface &$date, $invert = false, $parts = nu
         if (! $invert) {
             if ($originalHour >= $target) {
                 $distance = 24 - $originalHour;
-                $date = $this->timezoneSafeModify($date, "+{$distance} hours");
+                $date = $date->add(new \DateInterval("PT{$distance}H"));
 
                 $actualDay = (int)$date->format('d');
                 $actualHour = (int)$date->format('H');
                 if (($actualDay !== ($originalDay + 1)) && ($actualHour !== 0)) {
                     $offsetChange = ($previousOffset - $date->getOffset());
-                    $date = $this->timezoneSafeModify($date, "+{$offsetChange} seconds");
+                    $date = $date->add(new \DateInterval("PT{$distance}S"));
                 }
 
                 $originalHour = (int)$date->format('H');
             }
 
             $distance = $target - $originalHour;
-            $date = $this->timezoneSafeModify($date, "+{$distance} hours");
+            $date = $date->add(new \DateInterval("PT{$distance}H"));
         } else {
             if ($originalHour <= $target) {
                 $distance = ($originalHour + 1);
-                $date = $this->timezoneSafeModify($date, "-" . $distance . " hours");
+                $date = $date->sub(new \DateInterval("PT{$distance}H"));
 
                 $actualDay = (int)$date->format('d');
                 $actualHour = (int)$date->format('H');
                 if (($actualDay !== ($originalDay - 1)) && ($actualHour !== 23)) {
                     $offsetChange = ($previousOffset - $date->getOffset());
-                    $date = $this->timezoneSafeModify($date, "+{$offsetChange} seconds");
+                    $date = $date->add(new \DateInterval("PT{$offsetChange}H"));
```

Comment:
Can we be sure that $offsetChange is always positive? Because if it will be negative an error will occur here